src/renderer/grid_renderer.rs

```diff
@@ -121,12 +123,19 @@ impl GridRenderer {
             let random_color = random_hsv.to_color(255);
             paint.set_color(random_color);
         } else {
-            paint.set_color(style.background(&self.default_style.colors).to_color());
+            paint.set_color(style_background);
         }
-        if style.blend > 0 {
-            paint.set_alpha_f((100 - style.blend) as f32 / 100.0);
+
+        // Only apply transparency on the default background color
+        if style_background == self.get_default_background() {
+            if style.blend > 0 {
+                paint.set_alpha_f((100 - style.blend) as f32 / 100.0);
+            } else {
+                paint.set_alpha_f(SETTINGS.get::<WindowSettings>().opacity);
```

Comment:
Since you now use `opacity` and `text_backgrond_color` when preparing the lines, the setting will not properly refresh everything anymore.

This can be observed if setting `g:neovide_opacity=0.5` and then back `g:neovide_opacity=1`.  Some of the lines will not update properly.

The solution is to call `renderer.prepare_lines(true)` as a response to the `WindowSettingsChanged::Opacity` and `WindowSettingsChanged::text_background_opacity`

Reply:
Added the changes in 9448755